### PR TITLE
PLATFORM-1066 - Adds staging banner

### PR DESCRIPTION
**Pull request by kierangillen**
- Adds a purple banner to staging to make it more obvious
- https://artsyproduct.atlassian.net/browse/PLATFORM-1066

screen shot 2019-01-10 at 3 15 05 pm

### CODE REVIEW

File: src/desktop/components/main_layout/header/templates/index.jade [hunk 1]
```diff
@@ -1,5 +1,8 @@
 include ../../../util/activator
 
 header#main-layout-header( class= sd.HEADER_CLASS ? sd.HEADER_CLASS : '' )
+  if sd.APPLICATION_NAME === "force-staging"
+    #staging-banner
+      != stitch.components.StagingBanner()
```
Comment (damassi) [inline src/desktop/components/main_layout/header/templates/index.jade]: Nice, you found your way into stitch 👍

File: src/desktop/components/react/stitch_components/StagingBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import React from "react"
+import { Banner } from "@artsy/palette"
+
+export const StagingBanner = () => (
+  <Banner backgroundColor="purple100" message={"staging.artsy.net"} />
```
Comment (damassi) [inline src/desktop/components/react/stitch_components/StagingBanner.tsx]: Dont need brackets (`{}`) around strings

File: src/desktop/components/react/stitch_components/StagingBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import React from "react"
+import { Banner } from "@artsy/palette"
+
+export const StagingBanner = () => (
+  <Banner backgroundColor="purple100" message="staging.artsy.net" />
```
Comment (joeyAghion) [inline src/desktop/components/react/stitch_components/StagingBanner.tsx]: Feels slightly icky to hard-code a domain name that may or not match the location here based on an different config. Can we get this from express somehow instead?

Comment (damassi) [reply src/desktop/components/react/stitch_components/StagingBanner.tsx]: `sd.APP_URL`

**Comment by damassi** [comment]
One possibility would be to show the banner for a set period of time, and then hide it: 
```js
export class StagingBanner extends React.Component {
  state = {
    show: true
  }
  
  componentDidMount() {
    setTimeout(() => {
      this.setState({ show: false })
    }, VISIBLE_DURATION)
  }

  render() {
    return (
      if (!this.state.show) {
        return null
      }

      return ... 
    )
  }
}
```
When transitioning to different pages it will appear again for a short period so devs will remain aware of the env.

**Comment by kierangillen** [comment]
> One possibility would be to show the banner for a set period of time, and then hide it:
> 
> ```js
> export class StagingBanner extends React.Component {
>   state = {
>     show: true
>   }
>   
>   componentDidMount() {
>     setTimeout(() => {
>       this.setState({ show: false })
>     }, VISIBLE_DURATION)
>   }
> 
>   render() {
>     return (
>       if (!this.state.show) {
>         return null
>       }
> 
>       return ... 
>     )
>   }
> }
> ```
> 
> When transitioning to different pages it will appear again for a short period so devs will remain aware of the env.

I at first was going to make it dismissible but Brian thought it would be better to not be. Happy to add the duration but I could see that making this pointless if admins are using multiple tabs and switching back and forth. I could add an event for the tab being reopened that displays the banner again. But sounds more complicated than it's worth probably.

**Comment by joeyAghion** [comment]
Merging!

I still wonder if this may get in the way of using staging for production-like screenshots.

**Pull request merged**